tests/unit/data_classes/test_kinesis_firehose_response.py

```diff
@@ -0,0 +1,69 @@
+from aws_lambda_powertools.utilities.data_classes import (
+    KinesisFirehoseDataTransformationRecord,
+    KinesisFirehoseDataTransformationRecordMetadata,
+    KinesisFirehoseDataTransformationResponse,
+    KinesisFirehoseEvent,
+)
+from tests.functional.utils import load_event
+
+
+def test_kinesis_firehose_response():
```

Comment:
tip: use GIVEN/WHEN/THEN to help reviewers pick up the intent of the test.

For example, reading this code, I noticed we're deserializing b64 to str, then back again. If the intent is to test the Response record serializing correctly with the input data, we can pass `data=record.data` directly :) 

Then have a separate test where we transform the data to confirm it differs from the original record